cloudfront/cloudfront.go

```diff
@@ -0,0 +1,130 @@
+package cloudfront
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/YaleSpinup/s3-api/common"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+	"github.com/aws/aws-sdk-go/service/cloudfront/cloudfrontiface"
+	uuid "github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// CloudFront is a wrapper around the aws cloudfront service with some default config info
+type CloudFront struct {
+	Service         cloudfrontiface.CloudFrontAPI
+	Domains         map[string]common.Domain
+	WebsiteEndpoint string
+}
+
+// NewSession creates a new cloudfront session
+func NewSession(account common.Account) CloudFront {
+	c := CloudFront{}
+	log.Infof("creating new session with key id %s in region %s", account.Akid, account.Region)
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: credentials.NewStaticCredentials(account.Akid, account.Secret, ""),
+		Region:      aws.String(account.Region),
+	}))
+	c.Service = cloudfront.New(sess)
+	c.Domains = account.Domains
+	c.WebsiteEndpoint = "s3-website-" + account.Region + ".amazonaws.com"
+	return c
+}
+
+// ValidateWebsite validates the name of the website, ensuring we have a cert for the domain.  It splits the website
+// name in 2 pieces since we are using a wildcard.  This would need to change if we supported certificates per website.
+func (c *CloudFront) WebsiteDomain(name string) (*common.Domain, error) {
```

Comment:
Is this supposed to be called `ValidateWebsite` or `WebsiteDomain`?

Reply:
good catch, I renamed it when I started returning the domain